src/app/pages/apps/components/installed-apps/installed-apps.component.ts

```diff
@@ -116,21 +116,26 @@ export class InstalledAppsComponent implements OnInit, AfterViewInit {
     return this.selection.selected;
   }
 
+  get checkedApps(): App[] {
+    return this.checkedAppsNames
+      .map((name) => this.dataSource.find((app) => app.name === name))
+      .filter(Boolean);
```

Comment:
Why is this `filter` added here?

Reply:
> Why is this `filter` added here?

To tripple check that it mapped correctly and no undefined is there.

Reply:
In my opinion, adding general checks like this can hide other bugs. We should fix the issue where an undefined value is possible in this case instead of adding generic checks like this.

Reply:
> In my opinion, adding general checks like this can hide other bugs. We should fix the issue where an undefined value is possible in this case instead of adding generic checks like this.

there are no other bugs, I can just remove that if you want. It's just a extra check.